sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeUtils.scala

```diff
@@ -50,7 +50,10 @@ object DateTimeUtils {
 
   val TIMEZONE_OPTION = "timeZone"
 
-  def getZoneId(timeZoneId: String): ZoneId = ZoneId.of(timeZoneId, ZoneId.SHORT_IDS)
+  def getZoneId(timeZoneId: String): ZoneId = {
+    // To support the (+|-)h:mm format because it was supported before Spark 3.0.
+      ZoneId.of(timeZoneId.replaceFirst("(\\+|\\-)(\\d)(:)", "$10$2$3"), ZoneId.SHORT_IDS)
```

Comment:
Nit: unindent this line two spaces
Capturing the colon as a group isn't needed. Just match it directly and write a literal colon in the replacement.
The regex looks right; I'm just wondering if it can possibly accidentally match something else in the TZ. I can't think of it.

Reply:
didn't find anything wrong so far.